config-xml/addon/src/main/java/org/jboss/windup/config/parser/xml/MetadataRootHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.jboss.windup.config.parser.xml;
+
+import static org.joox.JOOX.$;
+
+import java.util.List;
+
+import org.jboss.forge.furnace.addons.AddonId;
+import org.jboss.windup.config.exception.ConfigurationException;
+import org.jboss.windup.config.metadata.MetadataBuilder;
+import org.jboss.windup.config.metadata.RulesetMetadata;
+import org.jboss.windup.config.metadata.TechnologyReference;
+import org.jboss.windup.config.parser.ElementHandler;
+import org.jboss.windup.config.parser.NamespaceElementHandler;
+import org.jboss.windup.config.parser.ParserContext;
+import org.w3c.dom.Element;
+
+@NamespaceElementHandler(elementName = "metadata", namespace = "http://windup.jboss.org/v1/xml")
+public class MetadataRootHandler implements ElementHandler<RulesetMetadata>
+{
+
+    @Override
+    public RulesetMetadata processElement(ParserContext context, Element element) throws ConfigurationException
+    {
+        List<Element> children = $(element).children().get();
+        MetadataBuilder metadataBuilder = context.getBuilder().getMetadataBuilder();
+        for (Element child : children)
+        {
+            Object result = context.processElement(child);
+            
+            switch ($(child).tag())
```

Comment:
It would be nice to require source and target. Although I guess that would require a significant amount of work on windup-rulesets?